erpnext/controllers/buying_controller.py

```diff
@@ -541,6 +541,14 @@ def update_stock_ledger(self, allow_negative_stock=False, via_landed_cost_vouche
 						})
 					sl_entries.append(sle)
 
+					if d.from_warehouse:
```

Comment:
On submission, make sl entries for source warehouse first, then do for target warehouse. On cancellation do reverse.
https://github.com/frappe/erpnext/blob/develop/erpnext/stock/doctype/stock_entry/stock_entry.py#L619
https://github.com/frappe/erpnext/blob/develop/erpnext/stock/doctype/stock_entry/stock_entry.py#L647